### PR TITLE
Adv protocol documentation

### PR DESCRIPTION
**Pull request by fraborg**
# Description

This PR adds some documentation for the advertisement protocol module.
There's a main file describing the protocol and its components: these are detailed in other files, one for the broacaster and one for the controller.

The labeler for documentation has been added too

### CODE REVIEW

File: docs/design/advertisement_protocol.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+3. The broadcaster retrieves the available cluster resources and, after applying some policies, creates an Advertisement.
+4. The Advertisement is pushed to the foreign cluster.
+  
+#### Ingoing chain
+1. An Advertisement is received from the foreign cluster
+2. The Advertisement is checked by a policy block
+3. If the Advertisement is accepted it is further processed by the controller
+4. The controller creates a virtual node with the information taken by the Advertisement
+
+### CRD generation
```
Comment (palexster) [inline docs/design/advertisement_protocol.md:34]: I would move this away.

File: .github/labeler.yml [hunk 2]
```diff
@@ -7,6 +7,9 @@ community:
   - LICENSE
   - README.md
 
+documentation:
+  - docs/**/*
```
Comment (palexster) [inline .github/labeler.yml:11]: Thanks!

File: docs/design/architecture.md [hunk 3]
```diff
@@ -10,11 +10,13 @@ Sharing resources with Liqo relies on three different phases:
 
 1. **Discovery**: The cluster looks for available clusters where offload new resources (e.g. neighborhood, dns, manual 
 insertion) and exchange credentials with each other to start communicate.
-2. **Advertisement management**: Clusters shares updates about the resourcing they are willing to export and their
+2. **Advertisement protocol**: Clusters shares updates about the resourcing they are willing to export and their
  capabilities (i.e. Advertisements)
 3. **Resource Sharing**: When a cluster is interested in resources proposed by a certain advertisement, it accepts the
  advertisement. This triggers the establishment of network interconnections and the spawning of a new virtual-kubelet.
 
+[Here](../images/complete-workflow.png) you can find a graphic description of the complete workflow
```
Comment (fraborg) [inline docs/design/architecture.md:18]: I added this picture of  the complete workflow. It's not very appealing, but I think it helps to understand the global behaviour. Tell me if it's ok, remove or modify it

Comment (palexster) [reply docs/design/architecture.md:18]: I think that picture is ok and very helpful

File: docs/design/advertisement_protocol.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+![architecture](../images/advertisement-protocol/architecture.png)
+
+### Components
+* The [broadcaster](broadcaster.md) is in charge of sending to other clusters the Advertisement message, containing the resources made available for sharing and (optionally) their prices
+* The [advertisement operator](controller.md) (briefly called **controller**) is the module that receives Advertisement messages and creates the virtual nodes with the announced resources.
+  
+### Workflow
+
+#### Outgoing chain
+1. A foreign cluster is provided (manually by the admin or through the [discovery protocol](discovery.md)).
```
[the remark this answers is not in the document]
Comment (fraborg) [reply docs/design/advertisement_protocol.md:23]: yeah, you're right
I wanted to give a 'high-level' view, hiding the internal logic (which the user doesn't see)
I updated it, tell me if it's better to further modify

[the remark this answers is not in the document]
Comment (aleoli) [reply docs/design/advertisement_protocol.md:23]: yes, I understood but the term could be confusing, it's easier to understand now

**Pull request merged**